hazelcast/src/main/java/com/hazelcast/map/impl/operation/MergeBackupOperation.java

```diff
@@ -0,0 +1,117 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.map.impl.MapEntries;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.map.impl.record.RecordInfo;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.BackupOperation;
+import com.hazelcast.spi.PartitionAwareOperation;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+import com.hazelcast.util.Clock;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.map.impl.EntryViews.createSimpleEntryView;
+import static com.hazelcast.map.impl.record.Records.applyRecordInfo;
+
+/**
+ * Creates backups for merged {@link MapEntries} after split-brain healing with a {@link SplitBrainMergePolicy}.
+ *
+ * @since 3.10
+ */
+public class MergeBackupOperation extends MapOperation implements PartitionAwareOperation, BackupOperation {
+
+    private MapEntries entries;
+    private List<RecordInfo> recordInfos;
+    private boolean hasWanReplication;
+
+    public MergeBackupOperation(String name, MapEntries entries, List<RecordInfo> recordInfos, boolean hasWanReplication) {
+        super(name);
+        this.entries = entries;
+        this.recordInfos = recordInfos;
+        this.hasWanReplication = hasWanReplication;
+    }
+
+    public MergeBackupOperation() {
+    }
+
+    @Override
+    public void run() {
+        for (int i = 0; i < entries.size(); i++) {
+            Data dataKey = entries.getKey(i);
+            Data dataValue = entries.getValue(i);
+            if (dataValue == null) {
+                recordStore.removeBackup(dataKey);
+                if (hasWanReplication) {
+                    mapEventPublisher.publishWanReplicationRemoveBackup(name, dataKey, Clock.currentTimeMillis());
+                }
+            } else {
+                Record record = recordStore.putBackup(dataKey, dataValue);
+                applyRecordInfo(record, recordInfos.get(i));
+                if (hasWanReplication) {
+                    Data dataValueAsData = mapServiceContext.toData(dataValue);
+                    EntryView entryView = createSimpleEntryView(dataKey, dataValueAsData, record);
+                    mapEventPublisher.publishWanReplicationUpdateBackup(name, entryView);
+                }
+                evict(dataKey);
```

Comment:
bulk evictions can add latency, we can quit if recordstore is at evictable threshold as in [here](https://github.com/hazelcast/hazelcast/blob/b97fff122712eeb1a26137063f00a9c2c1439850/hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java#L203)

Reply:
Do you mean I should replace this with:
```
                if (recordStore.shouldEvict()) {
                    evict(dataKey);
                }
```
?

Reply:
yes, i do.

Reply:
Okay, done.